### PR TITLE
Add Convert.FromHex benchmark

### PR DESCRIPTION
**Pull request by hrrrrustic**
See https://github.com/dotnet/runtime/pull/86556#issuecomment-1556276521

### CODE REVIEW

**Review by danmoseley — DISMISSED** [review]
not sure whether at some future point we might want to add a range of lengths or not -- more benchmarks has their cost. this is something.

File: src/benchmarks/micro/libraries/System.Runtime.Extensions/Perf.Convert.cs [hunk 1]
```diff
@@ -18,6 +18,7 @@ public class Perf_Convert
         private byte[] _binaryData;
         private char[] _base64CharArray;
         private string _base64String;
+        private string _hexString;
```
Comment (danmoseley) [inline src/benchmarks/micro/libraries/System.Runtime.Extensions/Perf.Convert.cs:21]:
```suggestion
#if NET5_0_OR_GREATER
        private string _hexString;
#endif
```

**Review by danmoseley — APPROVED** [review]

**Comment by danmoseley** [comment]
@LoopedBard3 fyi this failed to download the CLI, so I added some robustification. (Oddly, it needs the retries when I run locally today.)

**Review by danmoseley — APPROVED** [review]

**Comment by danmoseley** [comment]
@LoopedBard3 looks like someone other than me needs to sign off -- can you, and merge?

**Review by LoopedBard3 — APPROVED** [review]
LGTM

**Review by LoopedBard3 — APPROVED** [review]
LGTM

**Pull request merged**